core/src/toga/widgets/activityindicator.py

```diff
@@ -22,11 +24,13 @@ def __init__(
         """
         super().__init__(id=id, style=style)
 
-        self._impl = self.factory.ActivityIndicator(interface=self)
-
         if running:
             self.start()
 
+    def _create(self) -> None:
+        self.factory = get_platform_factory()
```

Comment:
> Why move the factory to here? Since every widget will need a factory, and we're in control of when `_create()` is invoked, I'm not sure I see why it makes sense to set `self.factory`, rather than setting it in the base widget and then assuming it exists in `_create()`.

Because different widgets — in particular third-party widgets — need to be able to customize where they get their implementations from. BitmapView, for instance, rolls its own factory, and Chart doesn't even need one.

> Also - would it make sense to make the contract of `_create()` _return_ the `_impl`? A backend might define other attributes, but if we require that "the" impl is returned, we can guarantee that every widget _will_ have an attribute named `_impl`, rather than relying on the backend to use the "right" name.

That makes a lot of sense, I like it.

Reply:
> Because different widgets — in particular third-party widgets — need to be able to customize where they get their implementations from. BitmapView, for instance, rolls its own factory, and Chart doesn't even need one.

That's true; however, in the Chart case, having the factory around doesn't hurt; and in the BitmapView case, right now it can be overwritten, and the long term game is to allow widgets to register themselves, so the system default factory *will* be useful. Moving it to the `_create()` methods means a lot of code churn to add it, and then take it out again in the future when we can rely on a system wide factory.

Reply:
While we're creating a new API requirement: how do you feel about the name `_create`? I just pulled it from the existing code of NumberInput and PasswordInput, but I'm starting to think `_create_impl` (or something similar) would be clearer.

Reply:
I guess my counter would be "what else are you planning to create"? It's pretty much an existential method for the widget; on that basis, I don't mind `_create()` as a method name. 

When I was reviewing, my first reaction was whether we should promote this to a public API - but I convinced myself that wasn't a bridge we needed to cross here. If it turns out there's value in making the method public, we can do that in the future; but the counterexample - that it's a bad idea to be public, because we need to change it as an API - is easy to walk back if it's private.

Reply:
That makes sense — I was probably just overthinking it.